### DISCUSSION TITLE
Can we translate from English to other languages? And error UnicodeEncodeError: 'charmap' codec can't encode character '\u015f' in position 26

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [openai/whisper#76] [Q&A]
Hello. I wonder if we can translate into another language? Also can we transcribe and translate at the same?

I tried like this but it didn't write any output due to an error

`whisper "D:\86 se courses youtube kanali\yazilim_muhendisligi_ders_1.mp3" --language tr --model base --device cuda --task translate --task transcribe`

The error message

```
Traceback (most recent call last):
  File "C:\Python399\Scripts\whisper-script.py", line 33, in <module>
    sys.exit(load_entry_point('whisper==1.0', 'console_scripts', 'whisper')())
  File "C:\Python399\lib\site-packages\whisper\transcribe.py", line 291, in cli
    print(result["text"], file=txt)
  File "C:\Python399\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u015f' in position 26: character maps to <undefined>
```

![image](https://user-images.githubusercontent.com/19240467/192012501-b5b457c5-b753-43b8-a59a-e97fdaaa9e6a.png)

**Comment by FurkanGozukara** [comment]
I have fixed error with manually updating scripts to the latest commit

But both transcribe and translate and saving them to the files at the same time I still couldn't achieve that

**Reply by DrMemoryFish** [reply]
I'm running Whisper locally, have you managed to translate/transcribe from English to another language?

**Comment by FurkanGozukara** [comment]
Transcription is not saved to a file when I use Turkish language 

testing with output directory at the moment

`whisper "D:\86 se courses youtube kanali\ders1.mp3" --language tr --model small --device cuda -o "c:\temp"`

edit : after providing output directory it worked

now trying to achieve both translate and transcribe at the same time

**Comment by dehma01** [comment]
Did you succeed to translate from english to another language in a nice way? Because I have found a little hack but I 'd rather go for a clean approach

**Reply by FurkanGozukara** [reply]
I couldn't find any way at all. Could you please share your little hack method? I would appreciate a lot. You can also email me if you want to keep it private : furkangozukara@gmail.com

**Reply by RicardoAHS** [reply]
this is my method:

!whisper "example.mp3" --model medium --language (The language you want to translate into)

**Reply by FurkanGozukara** [reply]
I see the logic. Thank you very much @RicardoAHS I will test it

**Reply by FurkanGozukara** [reply]
> this is my method:
> 
> !whisper "example.mp3" --model medium --language (The language you want to translate into)

I tested and it didnt work :( put tr to english audio still generated english

**Reply by RicardoAHS** [reply]
Did you try with Google Colab?

Install this and try again:

! pip install git+https://github.com/openai/whisper.git

I have only translated into Spanish, I don't know if that has anything to do with it.

**Reply by FurkanGozukara** [reply]
Google Colab pip install fails for me. why do we need google colab? Also to put ! in front of command I need google colab? 


`    pandas\_libs/algos.c(158842): error C2039: 'tp_print': is not a member of '_typeobject'
    C:\Python399\include\cpython/object.h(193): note: see declaration of '_typeobject'
    pandas\_libs/algos.c(158850): error C2039: 'tp_print': is not a member of '_typeobject'
    C:\Python399\include\cpython/object.h(193): note: see declaration of '_typeobject'
    pandas\_libs/algos.c(158855): error C2039: 'tp_print': is not a member of '_typeobject'
    C:\Python399\include\cpython/object.h(193): note: see declaration of '_typeobject'
    pandas\_libs/algos.c(158870): error C2039: 'tp_print': is not a member of '_typeobject'
    C:\Python399\include\cpython/object.h(193): note: see declaration of '_typeobject'
    pandas\_libs/algos.c(158883): error C2039: 'tp_print': is not a member of '_typeobject'
    C:\Python399\include\cpython/object.h(193): note: see declaration of '_typeobject'
    pandas\_libs/algos.c(164747): warning C4996: '_PyUnicode_get_wstr_length': deprecated in 3.3
    pandas\_libs/algos.c(164763): warning C4996: '_PyUnicode_get_wstr_length': deprecated in 3.3
    pandas\_libs/algos.c(167625): warning C4996: 'PyUnicode_FromUnicode': deprecated in 3.3
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2`

**Reply by FurkanGozukara** [reply]
> ! pip install git+https://github.com/openai/whisper.git

I have made it work on https://colab.research.google.com/

it really outputs translated to TR but it is so bad :D why could be translation is so bad?

**Reply by dehma01** [reply]
I realized that the model will automatically translate speech that is not in the language detected.
For example, if you have an audio in english and you want to translate it in turkish, you just have to add in the beginning of the audio some turkish speech.
Then, when you will run the model, it will detect turkish as main language and will translate the english content in turkish.

**Reply by FurkanGozukara** [reply]
> I realized that the model will automatically translate speech that is not in the language detected. For example, if you have an audio in english and you want to translate it in turkish, you just have to add in the beginning of the audio some turkish speech. Then, when you will run the model, it will detect turkish as main language and will translate the english content in turkish.

So it basically ignores our input. Can we force it without adding some turkish speech to the audio. because that would break all the timing.

**Reply by luiseduardo7code** [reply]
Ricardo, me puedes explicar por favor qué comando debo utilizar para que me transcriba un audio del inglés al español?

**Reply by JackCloudman** [reply]
some update about is?

**Reply by RicardoAHS** [reply]
> Ricardo, me puedes explicar por favor qué comando debo utilizar para que me transcriba un audio del inglés al español?

Hola, yo lo ejecuté en un espacio de trabajo de Google Colab, solo añade este comando para tener listo Whisper:

`! pip install git+https://github.com/openai/whisper.git`

El código que usé para hacer la traducción al español es el siguiente:

`!whisper "yourFile.mp3" --model medium --language Spanish`

La AI detecta como lenguaje principal el español, aunque esté en otro idioma, por lo que hace una traducción muy buena.

Demora un poco en generar los archivos .srt o .vtt, solo es cuestión de esperar.

Espero te funcione :D

**Reply by Botato300** [reply]
> > Ricardo, me puedes explicar por favor qué comando debo utilizar para que me transcriba un audio del inglés al español?
> 
> Hola, yo lo ejecuté en un espacio de trabajo de Google Colab, solo añade este comando para tener listo Whisper:
> 
> `! pip install git+https://github.com/openai/whisper.git`
> 
> El código que usé para hacer la traducción al español es el siguiente:
> 
> `!whisper "yourFile.mp3" --model medium --language Spanish`
> 
> La AI detecta como lenguaje principal el español, aunque esté en otro idioma, por lo que hace una traducción muy buena.
> 
> Demora un poco en generar los archivos .srt o .vtt, solo es cuestión de esperar.
> 
> Espero te funcione :D

you are god

**Reply by flexeed** [reply]
> > Ricardo, me puedes explicar por favor qué comando debo utilizar para que me transcriba un audio del inglés al español?
> 
> Hola, yo lo ejecuté en un espacio de trabajo de Google Colab, solo añade este comando para tener listo Whisper:
> 
> `! pip install git+https://github.com/openai/whisper.git`
> 
> El código que usé para hacer la traducción al español es el siguiente:
> 
> `!whisper "yourFile.mp3" --model medium --language Spanish`
> 
> La AI detecta como lenguaje principal el español, aunque esté en otro idioma, por lo que hace una traducción muy buena.
> 
> Demora un poco en generar los archivos .srt o .vtt, solo es cuestión de esperar.
> 
> Espero te funcione :D

Hola! Gracias por las instrucciones. 
Tengo un problema, mi audio esta en ingles y quiero transcribirlo al español, ejecuto las celdas como indicas, y empezó a traducir bien una parte del audio al español, pero luego simplemente siguió transcribiendo en ingles. Por ejemplo, venia transcribiendo "Hola, mi nombre es Juan, how are you?" y así siguió haciendolo en ingles todo el audio. ¿Sabes que puede ser y como solucionarlo? 
Muchas gracias.

**Reply by fclarat** [reply]
Yo lo cambie al modelo large y me anduvo bien. 

`!whisper "audio.mp3" --model large --language Spanish`
